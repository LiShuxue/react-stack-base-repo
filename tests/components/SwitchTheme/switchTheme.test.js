import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import SwitchTheme from 'components/switchTheme';

test('should render Header component correctly', () => {
  const { asFragment } = render(<SwitchTheme />);
  expect(asFragment()).toMatchSnapshot();
});

test('should have "dark theme" button and click this button should change to dark theme', () => {
  const { getByText } = render(<SwitchTheme />);

  const darkThemeBtn = getByText('dark theme');
  expect(darkThemeBtn).toBeInTheDocument();

  fireEvent.click(darkThemeBtn);
  expect(localStorage.getItem('theme')).toEqual('dark');
});

test('should have "light theme" button and click this button should change to light theme', () => {
  const { getByText } = render(<SwitchTheme />);

  const lightThemeBtn = getByText('light theme');
  expect(lightThemeBtn).toBeInTheDocument();

  fireEvent.click(lightThemeBtn);
  expect(localStorage.getItem('theme')).toEqual('light');
});
