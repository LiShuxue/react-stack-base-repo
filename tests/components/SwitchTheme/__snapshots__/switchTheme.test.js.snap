// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component correctly 1`] = `
<DocumentFragment>
  <button>
    dark theme
  </button>
  <button>
    light theme
  </button>
</DocumentFragment>
`;
